XIVComboExpanded/Combos/NIN.cs

```diff
@@ -320,3 +320,60 @@ protected override uint Invoke(uint actionID, uint lastComboActionID, float comb
         return actionID;
     }
 }
+
+internal class KassatsuTrickDream : CustomCombo
+{
+    protected internal override CustomComboPreset Preset { get; } = CustomComboPreset.NinjaDreamyTrickAttack;
+    protected override uint Invoke(uint actionID, uint lastComboActionID, float comboTime, byte level)
+    {
+        if (actionID == NIN.Kassatsu)
+        {
+            if (!IsCooldownUsable(NIN.Kassatsu) && HasEffect(NIN.Buffs.ShadowWalker))
+                return OriginalHook(NIN.TrickAttack);
+        }
+
+        if (actionID == NIN.Kassatsu && !IsCooldownUsable(NIN.Kassatsu) && !IsCooldownUsable(NIN.TrickAttack))
+            return NIN.DreamWithinADream;
+
+        return actionID;
+    }
+}
```

Comment:
I'd _strongly_ recommend implementing this into the existing block for Kassatsu that handles the existing Kassatsu->Trick Attack combo, and making the new 'Dreamy" combo a sub-combo rather than conflicting combo with the existing Kassatsu->Trick Attack combo.  Main reason is, the logic between the two is essentially already compatible, and multiple combo blocks using the same action ID are rather bug-prone, especially as logic is rewritten or expanded in the future.